src/main/java/de/qabel/desktop/daemon/management/AbstractBoxSyncBasedTransaction.java

```diff
@@ -3,11 +3,13 @@
 import de.qabel.desktop.config.BoxSyncConfig;
 import de.qabel.desktop.daemon.sync.event.ChangeEvent;
 import de.qabel.desktop.daemon.sync.event.WatchEvent;
+import de.qabel.desktop.nio.boxfs.BoxFileSystem;
+import de.qabel.desktop.nio.boxfs.BoxPath;
 import de.qabel.desktop.storage.BoxVolume;
 
 import java.nio.file.Path;
 
-public abstract class AbstractBoxSyncBasedTransaction extends AbstractTransaction {
+public abstract class AbstractBoxSyncBasedTransaction<S extends Path, D extends Path> extends AbstractTransaction<S, D> {
```

Comment:
That's why they are generics.
S (source) and D (destination) varies between Uploads and Downloads.
Uploads transfer from BoxPath to (local) Path and Downloads transfer from (local) Path to BoxPath. Kind of confusing but at least it is hidden for users of the Up- and Downloads.
```JAVA
public interface Upload extends Transaction<Path, BoxPath>
```